README.md

```diff
@@ -35,7 +35,125 @@ even when:
+
+1. Install `GatewayDNS` CRD on the management cluster
+   ```
+   kubectl apply -f "./manifests/crds/connectivity.tanzu.vmware.com_gatewaydns.yaml"
+   ```
+1. Install `capi-dns-controller` on the management cluster
+   ```
+   kubectl apply -f "./manifests/capi-dns-controller/deployment.yaml"
+   ```
+
+# Install Multi-cluster DNS on each workload cluster
+
+1. Deploy `dns-server` controller
+   ```
+   kubectl apply -f manifests/dns-server/
+   ```
+1. Get IP address assigned to the DNS server service
+   ```
+   kubectl get service -n capi-dns dns-server -o=jsonpath='{.spec.clusterIP}'
+   ```
+1. Patch CoreDNS to forward `xcc.test` to the `dns-server` server service's IP addr.
+   Note: the change is to add forwarding configuration for the `xcc.test` zone. Preserve your
+   Corefile's other configurations.
+   ```
+   kubectl -n kube-system patch configmap coredns \
+     --type=strategic --patch="$(cat <<EOF
+data:
+  Corefile: |
+    .:53 {
+        errors
+        health {
+           lameduck 5s
+        }
+        ready
+        kubernetes cluster.local in-addr.arpa ip6.arpa {
+           pods insecure
+           fallthrough in-addr.arpa ip6.arpa
+           ttl 30
+        }
+        prometheus :9153
+        forward . /etc/resolv.conf
+        cache 30
+        loop
+        reload
+        loadbalance
+    }
+    xcc.test {
+        forward . <REPLACE_WITH_IP_OBTAINED_IN_PRIOR_STEP>
+        reload
+    }
+EOF
+    )"
+    ```
+
+# Deploy a load balanced service to a workload `cluster-a`
+
+1. Install Contour
+   ```
+   kubectl apply -f manifests/contour/
+   ```
+1. Deploy a (kuard) workload
+   ```
+   kubectl apply -f manifests/example/kuard.yaml
+   ```
+
+# Lastly, wire up cross cluster connectivity
+
+1. On the management cluster, label the Clusters that have services that shall
+   be discoverable by other clusters. The `GatewayDNS` record created later will
+   use this label as a `ClusterSelector`.  In this example, the clusters are are
+   using the label `hasContour=true`.
+   ```
+   kubectl -n dev-team label cluster cluster-a hasContour=true --overwrite
+   kubectl -n dev-team label cluster cluster-b hasContour=true --overwrite
+   ```
+1. On the management cluster, create a GatewayDNS Record.
+   `kubectl -n dev-team apply -f manifests/example/dev-team-gateway-dns.yaml`
+
+   The GatewayDNS's spec has a `clusterSelector` that tells the
+   `capi-dns-controller` which clusters shall be watched for services. The
+   controller will look for a service with the namespace/name of `service`. In
+   this example, Contour runs a service in the `projectcontour/envoy`
+   namespace/name. The `resolutionType` this example's service is type
+   `loadBalancer`.
+   ```
+   ---
+   apiVersion: connectivity.tanzu.vmware.com/v1alpha1
+   kind: GatewayDNS
+   metadata:
+     name: dev-team-gateway-dns
+     namespace: dev-team
+   spec:
+     clusterSelector:
+       matchLabels:
+         hasContour: "true"
```

Comment:
I'm kind of thinking about changing our examples to use label like `gateway: contour`.  Thoughts?

Reply:
This will have an effect on e2e's, so we'll defer.